pkg/types/symboltable/symboltable.go

```diff
@@ -0,0 +1,419 @@
+package symboltable
+
+import (
+	"encoding/json"
+	"fmt"
+	gotypes "github.com/gofed/symbols-extractor/pkg/types"
+)
+
+type SymbolTable interface {
+	LookupSymbol(string) SymbolDef
+	//AddSymbol(string, SymbolDef) error
+	AddFunction(string, SymbolDef) error
+	AddVariable(string, SymbolDef) error
+	AddConstant(string, SymbolDef) error
+	AddDataType(string, SymbolDef) error
+}
+
+type DeclPos struct {
+	File string
+	Line uint
+}
+
+//NOTE: Def can be empty for variable, but not for the others;
+type SymbolDef struct {
+	Pos  DeclPos          `json:"-"`
+	Name string           `json:"name"`
+	Def  gotypes.DataType `json:"def,omitempty"`
+}
+
+
+func (sym *SymbolDef) GetPosition() DeclPos { return sym.Pos }
+func (sym *SymbolDef) GetName() string      { return sym.Name }
+func (sym *SymbolDef) GetType() string      { return sym.Def.GetType() }
```

Comment:
Those functions are not necessary as long as there is nothing like `Symbol` interface and the symbol is basically an attribute. We can access all fields directly as long as they are exported. E.g.

```go
s := &SymbolDef{
   Name: "Name",
   Pos: DeclPos{
      File: "file.go",
      Line: 55,
   },
   Def: nil,
}

fmt.Printf("Name: %v\n", s.Name)
fmt.Printf("Pos: %v\n", s.Pos)
fmt.Printf("Def: %#v\n", s.Def)
```